src/software/ai/evaluation/calc_best_shot_impl.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include "shared/constants.h"
+#include "software/ai/evaluation/shot.h"
+#include "software/geom/point.h"
+#include "software/geom/segment.h"
+#include "software/util/make_enum/make_enum.h"
+#include "software/world/field.h"
+#include "software/world/robot.h"
+#include "software/world/team.h"
+#include "software/world/world.h"
+
+/**
+ * Calculates the percentage of the net that a given shot is aiming for
+ *
+ * @param field The field the shot is being performed on
+ * @param shot_origin The point the shot is being taken from
+ * @param shot The shot being performed
+ * @param goal The goal to shoot at
+ *
+ * @return A value in [0,1] indicating the percentage of open net the shot is
+ *         being taken on. Larger numbers mean a larger percentage of the net is open
+ */
+double calcShotOpenNetPercentage(const Field &field, const Point &shot_origin,
+                                 const Shot &shot, TeamType goal);
+
+/**
+ * Function calculates the optimal shot and the corresponding Angle
+ * representing the 'open' area of that shot on a given segment to shoot at
+ *
+ * @param origin : The origin of the shot
+ * @param segment : The segment at which shots are being evaluated on
+ * @param obstacles : Any obstacle that can block the shot
+ *
+ * @return Shot : Returns the optimal Shot (Point and Angle) corresponding to the
+ * given parameters
+ * @return nullopt : A Shot does not exist
```

Comment:
can we clean up these weird `:`s?

Reply:
Was wondering about those. I agree with removing them.